### ISSUE TITLE
no errors but calendar buffers are still empty after fetch/sync

### ISSUE THREAD

**Issue by wminshew** [kidd/org-gcal.el#103]
sadly not getting any errors, but `fetch` and `sync` aren't working (the buffers are blank despite the messages buffer showing success, as logged below)

```
    (setq org-gcal-client-id "..."
          org-gcal-client-secret (get-authinfo ...)
          org-gcal-file-alist '(("x" . "~/development/org/gcal.org")
                                ("y" . "~/development/org/x-gcal.org")))
```

```
Completed event fetching .
Events fetched into
~/development/org/gcal.org
Events fetched into
~/development/org/x-gcal.org
```

I can see on google that they're receiving the request / responding w 200

any ideas how to debug this further?

**Comment by wminshew** [comment]
oddly enough it used to work. Not sure exactly when/why it stopped. Have tried deleting & re-installing the layers & underlying packages without luck, as well as refreshing the google token

**Comment by bensleveritt** [comment]
I'm seeing this too on a new installation

**Comment by FayeAlephNil** [comment]
I'm also experiencing this issue on a new installation

**Comment by pomeo** [comment]
Was the same, with no errors and empty buffer. In the README there is info about "(setq gnutls-algorithm-priority "NORMAL:-VERS-TLS1.3")". I added this line, deleted all created files and tokens, get a new token and all works.

**Comment by wminshew** [comment]
Switching to the `org-gcal` from melpa appears to have solved this for me, but I am going to leave the issue open bc the repo still appears broken

**Comment by nelson-liu** [comment]
Also seeing this issue, but using org-gcal from MELPA didn't work for me---it should be the same source as what's in this repo, no?

**Comment by swapnilsm** [comment]
This happened with me as well after the first sync. Then I tried creating a new event and it was getting pulled correctly. So looked like gcal was storing some token about the events it has pulled in already. I checked and found that running `org-gcal-sync-tokens-clear` will clear all those and will force fetching all the calendar events again.

**Comment by telotortium** [comment]
Yes, what @swapnilsm may help.

**Comment by wminshew** [comment]
my melpa workaround appeared to only work temporarily. Perhaps downloading org-gcal from a new source manually reset the sync-tokens.. regardless, `org-gcal-sync-tokens-clear` seems to help!

**Comment by Harjuu** [comment]
I had the same issue and tried `org-gcal-sync-tokens-clear`. Running `org-gcal-sync` afterwards gives me `deferred error : (wrong-type-argument hash-table-p nil)`. After fetching once the sync runs without errors, but the buffers are still empty. No difference when scheduling a new event, still nothing appearing in any of the two files.

Everything I've got in my config for gcal:
```
(require 'org-gcal)
(setq org-gcal-client-id "MYCLIENTID"
      org-gcal-client-secret "MYSECRET"
      org-gcal-fetch-file-alist '(("FIRSTCALENDARID" .  "~/personal.org")
                                  ("SECONDCALENDARID" .  "~/sch.org")))

```

**Comment by bensleveritt** [comment]
I noticed that my old org file was being updated, and after deleting it and clearing the tokens, it seems to be resolved for now.

**Comment by bmills22** [comment]
I receive the same error as @Harjuu. @bensleveritt's fix (deleting the org file and running `org-gcal-sync-tokens-clear`) worked once then never again :man_shrugging: 

I trigger this issue by deleting a calendar entry locally and running `org-gcal-sync`.

Other specs:
Doom Emacs (latest)
Emacs 27.1.50 (running as a daemon)
Ubuntu 20.04

**Comment by cvanelteren** [comment]
I would like to add to the pile. Refreshing tokens does not help. Syncing new appointments works, but fetching present once doesn't. 

Is there any update on a fix?

**Comment by swapnilsm** [comment]
I checked and found that gcal was trying to lookup for the file defined in
`org-id-locations-file` which was non-existent in my case. I created a
blank file at that location and then running `org-gcal-fetch` worked with
me.

Regards,
Swapnil Mahajan

**Comment by nelson-liu** [comment]
@swapnilsm 's fix worked for me as well---thanks for sharing!

It might be applicable to others like me who are running doom emacs. In my case, doom had set `org-id-locations-file`, but it / its folder was nonexistent.

**Comment by Harjuu** [comment]
@swapnilsm 's solution worked for me too! Also running doom emacs. Cheers

**Comment by grenmester** [comment]
I'm running Doom Emacs and I am encountering the same issue. I tried @swapnilsm's solution by creating an empty file at `~/.emacs.d/.org-id-locations` which is the default location for the `org-id-locations-file` and also running `org-gcal-sync-token-clear` but neither seem to work. My `org-gcal` config is basically the same as the ones shown above but it still doesn't seem to work for me. Does anyone have any ideas or potential solutions to this problem?

**Comment by swapnilsm** [comment]
@grenmester Does `fetch` result into some error for you?

**Comment by grenmester** [comment]
No, fetch runs successfully and creates the specified directories and files if they don’t exist but the files are empty as in the cases above.

**Comment by apc** [comment]
I just tried sorting out some issue with two calendars having different time zones and tried deleting one of the local files I was syncing with one of my calendars. I'm now also in a position of getting no errors when syncing or fetching, but getting an empty file when I do so too. Tried running `org-gcal-sync-token-clear` but that didn't help either.
